### PR TITLE
Adding product recognition to cognitive service python samples

### PR DESCRIPTION
**Pull request by lucas95123**
## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[ x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by pj-ms** [comment]
It could be confusing to have a folder for product reco and a folder for model customization, as product reco has model customization as well

**Comment by pj-ms** [comment]
need update for readme.md

**Comment by lucas95123** [comment]
> It could be confusing to have a folder for product reco and a folder for model customization, as product reco has model customization as well.
> 
> Many documentations such as quota limit, faq, aiq faq in model customization applies or requires update from PoS as well

Yes, and the model customization part of the product recognition is still in the model_customization folder, the product recognition only contains image stitching, product recognition prediction, planogram compliance. Does that work

**Comment by lucas95123** [comment]
> need update for readme.md

Gina will be working on that

File: cognitive_service_vision_model_customization_python_samples/clients/product_recognition_client.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import logging
+import time
+
+from cognitive_service_vision_model_customization_python_samples.clients import Client
+from ..models import ProductRecognition, ProductRecognitionResponse, ProductRecognitionStatus
+
+logger = logging.getLogger(__name__)
+
+
+class ProductRecognitionClient(Client):
```
[the remark this answers is not in the document]
Comment (lucas95123) [reply cognitive_service_vision_model_customization_python_samples/clients/product_recognition_client.py:10]: Training is the same TraningClient, this is for async product recognition predictoin

File: cognitive_service_vision_model_customization_python_samples/clients/product_recognition_client.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
+import logging
+import time
+
+from cognitive_service_vision_model_customization_python_samples.clients import Client
```
[the remark this answers is not in the document]
Comment (lucas95123) [reply cognitive_service_vision_model_customization_python_samples/clients/product_recognition_client.py]: fixed

File: cognitive_service_vision_model_customization_python_samples/clients/image_composition_client.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    def __init__(self, resource_type, resource_name: str, multi_service_endpoint, resource_key: str) -> None:
+        super().__init__(resource_type, resource_name, multi_service_endpoint, resource_key)
+
+    def stitch_images(self, request: ImageStitchingRequest) -> bytes:
+        json_response = self.request_post('/imagecomposition:stitch', data=request.to_json(), content_type='application/json')
+        return json_response.content
+    
+    def rectify_image(self, request: ImageRectificationRequest) -> bytes:
+        json_response = self.request_post('/imagecomposition:rectify', data=request.to_json(), content_type='application/json')
+        return json_response.content
```
[the remark this answers is not in the document]
Comment (lucas95123) [reply cognitive_service_vision_model_customization_python_samples/clients/image_composition_client.py]: fixed all

File: cognitive_service_vision_model_customization_python_samples/clients/client.py [hunk 4]
```diff
@@ -47,6 +49,10 @@ def request_post(self, path, params=None, data=None, content_type=None):
         params = params or {}
         headers = dict(self._headers, **{'Content-Type': content_type}) if content_type else self._headers
         r = requests.post(self._construct_url(path), data=data, params=dict(self._params, **params), headers=headers)
+
+        if r.headers['Content-Type'] == 'image/jpeg':
```
[the remark this answers is not in the document]
Comment (lucas95123) [reply cognitive_service_vision_model_customization_python_samples/clients/client.py:53]: Updated, for now our PUT API always return json, can extend in future if there's need

File: cognitive_service_vision_model_customization_python_samples/clients/product_recognition_client.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (pj-ms) [reply cognitive_service_vision_model_customization_python_samples/clients/product_recognition_client.py:10]: Are the create_run, query_run, delete_run and wait_for_completion for prediction?

**Review by pj-ms — APPROVED** [review]

**Comment by pj-ms** [comment]
pkg requres version update as well

File: docs/quota_limit.md [hunk 5]
```diff
@@ -2,35 +2,35 @@
...
-| Max image size                     | 20MB                            | 20MB                            |
-| Max image width/height             | 10,240                          | 10,240                          |
-| Min image width/height             | 10                              | 10                              |
-| Available regions                  | West US 2, East US, West Europe | West US 2, East US, West Europe |
-| Accepted image types               | jpg, png, bmp, gif, jpeg        | jpg, png, bmp, gif, jpeg        |
+|                                    | Generic IC                      | Generic OD                      | Product Recognition             |
+| ---------------------------------- | ------------------------------- | ------------------------------- | ------------------------------- |
+| Max # training hours*              | 336 (14 compute days)           | 1344 (56 compute days)          | 336 (14 compute days)           |
+| Max # training images              | 1,000,000                       | 200,000                         | 1,000,000                       |
```
Comment (pj-ms) [inline docs/quota_limit.md:8]: is product reco training with OD dataset?

Comment (pj-ms) [reply docs/quota_limit.md:8]: If so, 1M images could lead to like 10M images in the real IC training I suppose?

Comment (lucas95123) [reply docs/quota_limit.md]: Updated

**Pull request merged**